src/clj/rems/ext/duo.clj

```diff
@@ -70,13 +72,11 @@
    "DUO:0000027" {:restrictions [{:type :project}]}
    "DUO:0000028" {:restrictions [{:type :institute}]}})
 
-(def abstract-codes #{"DUO:0000001" "DUO:0000017"}) ; not concrete types can't be used as tag
-
-(defn- enrich-duo-code
-  "Adds convenience attributes to DUO codes."
-  [duo]
-  (merge duo
-         (get complex-codes (:id duo))))
+;; http://purl.obolibrary.org/obo/DUO_0000001 data use permission
+;; http://purl.obolibrary.org/obo/DUO_0000006 health or medical or biomedical research
+;; http://purl.obolibrary.org/obo/DUO_0000017 data use modifier
+;; http://purl.obolibrary.org/obo/DUO_0000042 general research use
+(def abstract-codes #{"DUO:0000001" "DUO:0000006" "DUO:0000017" "DUO:0000042"}) ; not concrete types can't be used as tag
```

Comment:
this list could be reviewed. the DUO code queries should be done at the most specific level, and only after that by checking ancestors. here `DUO_0000042` (GRU) and `DUO_0000006` (HMB) are ancestors of `DUO_0000007` (DS) so technically they could be used as DUO codes in a resource and allow validating application `DS` DUO code , but we discussed that these might be just abstract codes and not actually used codes.